### PR TITLE
enh(Resource-status/Filter): Revamp Filter

### PR DESCRIPTION
**Pull request by bdauria**
## Description

This changes the Filter behaviour to combine it with the search:

ezgif-7-5e8af7407726

**Fixes** # (issue)

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 21.10.x (master)

### CODE REVIEW

File: www/front_src/src/Resources/Filter/index.test.tsx [hunk 1]
```diff
@@ -333,7 +342,7 @@ describe(Filter, () => {
       },
     ],
   ])(
-    'executes a listing request with "%p" params when "%p" filter group is set',
+    'executes a listing request with "%p" params when "%p" filter is set',
```
Comment (Thebarda) [inline www/front_src/src/Resources/Filter/index.test.tsx:345]:
```suggestion
    'executes a listing request with "%p" parameters when "%p" filter is set',
```

File: www/front_src/src/Resources/Filter/useFilter.ts [hunk 2]
```diff
@@ -84,23 +89,21 @@ const useFilter = (): FilterState => {
...
   const [customFilters, setCustomFilters] = React.useState<Array<Filter>>([]);
-  const [filter, setFilter] = React.useState(getDefaultFilter());
-  const [nextSearch, setNextSearch] = React.useState<string | undefined>(
-    getDefaultSearchCriteria().value as string,
-  );
-  const [filterExpanded, setFilterExpanded] = React.useState(
-    getDefaultFilterExpanded(),
-  );
+  const [currentFilter, setCurrentFilter] = React.useState(getDefaultFilter());
+  const [appliedFilter, setAppliedFilter] = React.useState(getDefaultFilter());
```
Comment (Thebarda) [inline www/front_src/src/Resources/Filter/useFilter.ts:94]: If I understand correctly, the applied filter is the current filter when the user press the "Enter" key, right ?

File: www/front_src/src/Resources/translatedLabels.ts [hunk 3]
```diff
@@ -216,5 +216,6 @@ export const labelMaxDuration1Year = 'The duration must be lesser than a year';
 export const labelAsDisplayed = 'As displayed';
 export const labelMediumSize = 'Medium size';
 export const labelSmallSize = 'Small size';
+export const labelSearchOptions = 'Search options';
```
Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts:219]: Do not forget to edit the translation files

File: www/front_src/src/Resources/Filter/useFilter.ts (hunk 2 above)
Comment (bdauria) [reply www/front_src/src/Resources/Filter/useFilter.ts:94]: Yes, or if a Filter is selected; or the Search button is pressed

**Comment by Thebarda** [comment]
That is a big change !
I really like the "Centreon Query Language" and sort each criterias by their order looks a wise way. 
I have one remark for @sbrizini and @bdauria: I think the criterias icon is, in my opinion, too far away to the right. Maybe it gonna looks better to move to the left of the search bar. Or maybe the search bar is too width compare to the search on Gmail.

One more remark, when typing the beginning of a criteria in the search bar, like "state:", when expanding the criterias options, the state criteria tells that I have a State selected when it is not.
https://user-images.githubusercontent.com/12515407/125921085-1d7a46c0-1f82-4c4c-ab1d-79e7536c9eec.mov

**Comment by bdauria** [comment]
> That is a big change !
> I really like the "Centreon Query Language" and sort each criterias by their order looks a wise way.
> I have one remark for @sbrizini and @bdauria: I think the criterias icon is, in my opinion, too far away to the right. Maybe it gonna looks better to move to the left of the search bar. Or maybe the search bar is too width compare to the search on Gmail.
> 
> One more remark, when typing the beginning of a criteria in the search bar, like "state:", when expanding the criterias options, the state criteria tells that I have a State selected when it is not.
> https://user-images.githubusercontent.com/12515407/125921085-1d7a46c0-1f82-4c4c-ab1d-79e7536c9eec.mov

Thanks for the feedback. I think it'll be best handled in another step so that we don't clutter this one even more ;)

**Review by Thebarda — COMMENTED** [review]
Do not forget to update the translation files too :)

**Comment by bdauria** [comment]
> Do not forget to update the translation files too :)

Done ;)

**Review by Thebarda — APPROVED** [review]
Tested & LGTM

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**